File: mobile/src/domain/entities/lists.ts
export interface ListOwner {
  id: number;
  username: string;
  display_name: string | null;
  avatar_url: string | null;
}

export interface ListPermissions {
  can_edit: boolean;
  can_delete: boolean;
  can_manage_collaborators: boolean;
}

export interface MediaSummary {
  tmdb_id: number;
  media_type: 'movie' | 'tv';
  title: string | null;
  poster_path: string | null;
  release_date: string | null;
}

export interface ListSummary {
  id: number;
  name: string;
  description: string | null;
  is_public: boolean;
  owner: ListOwner;
  items_count: number;
  relationship: 'owner' | 'collaborator' | 'viewer';
  created_at: string;
  updated_at: string;
}

export interface ListItem {
  tmdb_id: number;
  media_type: 'movie' | 'tv';
  position: number;
  added_at: string;
  added_by: ListOwner;
  media_summary: MediaSummary | null;
}

export interface ListInvitation {
  id: number;
  list_id: number;
  list_name: string;
  list_description: string | null;
  list_is_public: boolean;
  owner: ListOwner;
  invited_by: ListOwner;
  created_at: string;
}

export interface MyListsOverview {
  owned_lists: ListSummary[];
  shared_lists: ListSummary[];
  pending_invitations_received: ListInvitation[];
}

export interface ListDetail extends ListSummary {
  collaborators: ListOwner[];
  permissions: ListPermissions;
  items: ListItem[];
}

export interface ListWriteRequest {
  name: string;
  description: string | null;
  is_public: boolean;
}

export interface AddListItemRequest {
  tmdb_id: number;
  media_type: 'movie' | 'tv';
}

export interface ListItemRef {
  tmdb_id: number;
  media_type: 'movie' | 'tv';
}

export interface ReorderListItemsRequest {
  source: ListItemRef;
  target: ListItemRef;
}

export interface CreateListInvitationRequest {
  invitee_user_id: number;
}

File: ENTREGA_TFG/Aplicacion/Back/app/presentation/schemas/lists.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


class ListUserResponse(BaseModel):
    id: int
    username: str
    display_name: str | None
    avatar_url: str | None


class MediaSummaryResponse(BaseModel):
    tmdb_id: int
    media_type: Literal["movie", "tv"]
    title: str | None
    poster_path: str | None
    release_date: str | None


class ListSummaryResponse(BaseModel):
    id: int
    name: str
    description: str | None
    is_public: bool
    owner: ListUserResponse
    items_count: int
    relationship: Literal["owner", "collaborator", "viewer"]
    created_at: datetime
    updated_at: datetime


class ListPermissionsResponse(BaseModel):
    can_edit: bool
    can_delete: bool
    can_manage_collaborators: bool


class ListItemResponse(BaseModel):
    tmdb_id: int
    media_type: Literal["movie", "tv"]
    position: int
    added_at: datetime
    added_by: ListUserResponse
    media_summary: MediaSummaryResponse | None


class ListDetailResponse(ListSummaryResponse):
    collaborators: list[ListUserResponse]
    permissions: ListPermissionsResponse
    items: list[ListItemResponse]


class ListInvitationResponse(BaseModel):
    id: int
    list_id: int
    list_name: str
    list_description: str | None
    list_is_public: bool
    owner: ListUserResponse
    invited_by: ListUserResponse
    created_at: datetime


class MyListsResponse(BaseModel):
    owned_lists: list[ListSummaryResponse]
    shared_lists: list[ListSummaryResponse]
    pending_invitations_received: list[ListInvitationResponse]


class CreateListRequest(BaseModel):
    name: str = Field(min_length=1, max_length=80)
    description: str | None = Field(default=None, max_length=280)
    is_public: bool = True


class UpdateListRequest(BaseModel):
    name: str = Field(min_length=1, max_length=80)
    description: str | None = Field(default=None, max_length=280)
    is_public: bool


class AddListItemRequest(BaseModel):
    tmdb_id: int
    media_type: Literal["movie", "tv"]


class ListItemRefRequest(BaseModel):
    tmdb_id: int
    media_type: Literal["movie", "tv"]


class ReorderListItemsRequest(BaseModel):
    source: ListItemRefRequest
    target: ListItemRefRequest


class CreateListInvitationRequest(BaseModel):
    invitee_user_id: int
